### PR TITLE
Added Save Load functions to Core.

### PR DESCRIPTION
**Pull request by ValeriiSudakov**
Saving is implemented by serializing the protobuf library.
TaskStorage has private methods "Serialize and Deserialize", that implement logic of saving/loading tasks data. These methods are called from Save/Load methods of this class.
TaskRepository also has Save/Load methods. Both call Save/Load methods of TaskStorage. Load method also fill TaskView with loading data.
TaskService: Save/Load methods just call same TaskRepository methods.
Added tests for it.

### CODE REVIEW

File: core/Memory_Model/Storage/TaskStorageClass.cpp [hunk 1]
```diff
@@ -52,4 +58,117 @@ bool TaskStorageClass::PostponeTask(const TaskID& id, const Date& date){
...
+  if (!file.is_open()){
+    return false;
+  }
+
+  storageToSave.SerializeToOstream(&file);
+  file.close();
+  return true;
+}
+
+SerializedTask::Priority PriorityToSerializedPriority(const Priority& priority){
```
Comment (denys-hrulov) [inline core/Memory_Model/Storage/TaskStorageClass.cpp:148]: Maybe ToSerialializedPriority will be enough comprehent?

File: core/Memory_Model/Storage/TaskStorageClass.cpp [hunk 2]
```diff
@@ -52,4 +58,117 @@ bool TaskStorageClass::PostponeTask(const TaskID& id, const Date& date){
...
+  for (const auto &task : tasks_) {
+    idMapping[task.first] = i;
+    i++;
+  }
+
+  SerializedStorage storageToSave;
+
+  for (const auto& task : tasks_){
+    auto* addTask = storageToSave.add_tasks();
+    auto* newTask = new SerializedTask;
```
Comment (denys-hrulov) [inline core/Memory_Model/Storage/TaskStorageClass.cpp]: No good reason for using raw pointer. Should use unique_ptr.

File: core/Memory_Model/Storage/TaskStorageClass.cpp [hunk 3]
```diff
@@ -52,4 +58,117 @@ bool TaskStorageClass::PostponeTask(const TaskID& id, const Date& date){
     return result;
   }
   return false;
+}
+
+bool TaskStorageClass::LoadFromFile(const std::string &fileName) {
+  if (fileName == ""){
```
Comment (denys-hrulov) [inline core/Memory_Model/Storage/TaskStorageClass.cpp:64]: Is it really neccessary?

**Review by denys-hrulov — CHANGES_REQUESTED** [review]
Found some inaccuracies

File: core/Memory_Model/Storage/TaskStorageClass.cpp (hunk 2 above)
Comment (ValeriiSudakov) [reply core/Memory_Model/Storage/TaskStorageClass.cpp:124]: Why? This pointer only need to fill the task. Protobuf is responsible to delete it.

Comment (denys-hrulov) [reply core/Memory_Model/Storage/TaskStorageClass.cpp]: Protobuf is responible for deleting addTask but not newTask.

**Review by denys-hrulov — CHANGES_REQUESTED** [review]
Still do not understand empty file path check

**Review by SeenKeen — APPROVED** [review]

**Review by denys-hrulov — APPROVED** [review]

**Pull request closed without merging**